src/meta/src/hummock/manager/mod.rs

```diff
@@ -1164,7 +1164,7 @@ where
             // The compaction task is finished.
             let mut versioning_guard = write_lock!(self, versioning).await;
             let versioning = versioning_guard.deref_mut();
-            let current_version = &mut versioning.current_version;
+            let mut current_version = versioning.current_version.clone();
```

Comment:
Why replace &mut with clone?

Reply:
Because we must make sure that this change can take effect only after writing ETCD successfully.

Reply:
The clone can be avoided for `report_compact_task_impl`, because the infallible (or panic) `apply_version_delta` is the only place that changes `current_version`.  But I agree making a clone helps to avoid bug caused by future change.